src/services/chatService.ts

```diff
@@ -38,21 +39,51 @@ export const getEmbeddings = async (query: string) => {
   return embeddings;
 };
 
+const generateFilter = (focus: Focus) => {
+  if (focus.name === 'core') {
+    const allowedScopes: string[] = [
+      'ucop',
+      'ucdppm',
+      'ucdppsm',
+      'ucddelegation',
+      'ucdinterim',
+    ];
+
+    return {
+      terms: {
+        'metadata.scope.keyword': allowedScopes,
+      },
+    };
+  } else if (focus.name === 'apm') {
+    return {
+      terms: {
+        'metadata.scope.keyword': ['ucdapm'],
+      },
+    };
+  } else if (focus.name === 'unions') {
+    // TODO: add union scopes and read variant
+    return {
+      terms: {
+        'metadata.scope.keyword': [],
+      },
+    };
+  } else {
+    // match nothing since we don't know what to do
+    return {
+      match_none: {},
+    };
+  }
+};
```

Comment:
typing on this whole thing would be nice. looks like the filter type should be `QueryDslQueryContainer | QueryDslQueryContainer[];` but allowedScopes as an enum or something would be nice